Service/DataService.php

```diff
@@ -2010,4 +2010,83 @@ public function createAndMapIndex(Environment $environment): void
             'name' => $environment->getAlias()
         ]);
     }
+
+    public function lockAllContentTypesRevisions(\DateTime $until, string $by): int
+    {
+        $rows = 0;
+        $contentTypes = $this->contentTypeService->getAll();
+        foreach ($contentTypes as $contentType) {
+            $rows += $this->lockContentTypeRevisions($contentType, $until, $by);
+        }
+
+        return $rows;
+    }
+
+    public function lockContentTypeRevisions(ContentType $contentType, \DateTime $until, string $by): int
+    {
+        $rows = 0;
+        try {
+            $rows += $this->revRepository->lockRevisions($contentType, $until, $by, true, false);
+        } catch (LockedException $e) {
+            $this->logger->error('service.data.lock_revisions_error', [
+                EmsFields::LOG_CONTENTTYPE_FIELD => $contentType->getName(),
+                EmsFields::LOG_USERNAME_FIELD => $by,
+                EmsFields::LOG_EXCEPTION_FIELD => $e,
+                EmsFields::LOG_ERROR_MESSAGE_FIELD => $e->getMessage(),
+            ]);
+        }
+
+        return $rows;
+    }
+
+    public function unlockAllContentTypesRevisions(string $by): int
+    {
+        $rows = 0;
+        $contentTypes = $this->contentTypeService->getAll();
+        foreach ($contentTypes as $contentType) {
+            $rows += $this->unlockContentTypeRevisions($contentType, $by);
+        }
+
+        return $rows;
+    }
+
+    public function unlockContentTypeRevisions(ContentType $contentType, string $by): int
+    {
+        $rows = 0;
+        try {
+            $rows += $this->revRepository->unlockContentTypeRevisionsbyUser($contentType, $by);
+        } catch (LockedException $e) {
+            $this->logger->error('service.data.unlock_revisions_error', [
+                EmsFields::LOG_CONTENTTYPE_FIELD => $contentType->getName(),
+                EmsFields::LOG_USERNAME_FIELD => $by,
+                EmsFields::LOG_EXCEPTION_FIELD => $e,
+                EmsFields::LOG_ERROR_MESSAGE_FIELD => $e->getMessage(),
+            ]);
+        }
+
+        return  $rows;
+    }
+
+    public function getAllContentTypesDraftedRevisions(): array
+    {
+        $draftedRevisions = [];
+        $contentTypes = $this->contentTypeService->getAll();
+        foreach ($contentTypes as $contentType) {
+            $draftedRevisions = \array_merge($draftedRevisions, $this->getContentTypeDraftedRevisions($contentType));
+        }
+
+        return $draftedRevisions;
+    }
+
+    public function getContentTypeDraftedRevisions(ContentType $contentType): array
+    {
+        return $this->revRepository->findBy([
```

Comment:
Move this to a custom repository function, with a good name. Suggestion findDraftsByContentType